libs/application/templates/parental-leave/src/lib/parentalLeaveUtils.ts

```diff
@@ -110,12 +110,12 @@ export function formatPeriods(
     let canDelete = startDateDateTime.getTime() > currentDateStartTime()
     const today = new Date()
 
-    if (applicationFundId === '') {
+    if (!applicationFundId || applicationFundId === '') {
       canDelete = true
     } else if (canDelete && today.getDate() >= 20) {
       const startDateBeginOfMonth = addDays(
         startDateDateTime,
-        startDateDateTime.getDate() * -1,
+        startDateDateTime.getDate() * -1 + 1,
```

Comment:
I find it a little confusing what this `startDateDateTime.getDate() * -1 + 1` represents. I would find it more clear if this end condition would be set to a variable with a descriptive name.